lib/cluster.js

```diff
@@ -260,6 +260,9 @@ function masterInit() {
       return process.nextTick(setupSettingsNT, settings);
     initialized = true;
     schedulingPolicy = cluster.schedulingPolicy;  // Freeze policy.
+    if (settings.scheduler) {
+      settings.scheduler(handles);
```

Comment:
@cjihrig the reason for not wanting to call the scheduler function every time something is distributed is because of performance reason. Basically for our use case we want to improve performance caused by the GC, so remove or add a work from the .free property which is used to distribute. In our use case, we know when need to run full GC before it starts causing latency, so the master cluster removes the work from the .free list and sends a message to the work to force full GC. Once the work is done with GC, it tells the master to put it back into the .free list. In a nutshell, the test script shows how the logic works.